src/chatbot/src/chatbot/aiml/DefaultSubs.py

```diff
@@ -48,8 +50,8 @@
     "hers": "his",
     "herself": "himself",
 }
-
-defaultPerson = {
+# These now match the comment above
+defaultPerson2 = {

```

Comment:
Hi Linas, I'll have another look at this.  I think it's a case of two bugs
then giving the right behavior.

For months I've seen problems and looked for alternative substitution rules
in the aiml.  I made up that banana example, but I've seen examples in the
last week in our testing transcripts where a first to second gets a first
to third.  I changed it because trying to chase this down I found a comment
above (as I noted) in that code suggesting that the person and person2 have
the opposite semantics than what you described.

Maybe there is a bug in the implementation of person and person2 handlers
w.r.t. to the AIML spec, and the right fix is to fix those handlers and
change the misleading comment.  I didn't go look at the spec, just
experimentally made the change and found that a long standing annoyance
appeared to be fixed.

On Sat, Aug 20, 2016 at 10:46 AM, Linas Vepštas <notifications@github.com>
wrote:

> In src/chatbot/src/chatbot/aiml/DefaultSubs.py
> <https://github.com/hansonrobotics/HEAD/pull/168#discussion_r75580623>:
>
> > @@ -48,8 +50,8 @@
> >      "hers": "his",
> >      "herself": "himself",
> >  }
> > -
> > -defaultPerson = {
> > +# These now match the comment above
> > +defaultPerson2 = {
>
> this looks incorrect. According to the AIML spec, person2 converts from
> 1st person to second person for example, I <--> you while the below is
> clearly for 3rd person ...
>
> Likewise, the person rule is supposed to convert 1st to 3rd person.
>
> I'm guessing that the "I like bannanas" rule is broken -- it is using
> person instead of person2 -- I suggest that the correct fix is to fix the
> rule, instead of the definition of the conversion.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/hansonrobotics/HEAD/pull/168/files/85ee952bbb4ef504d4b96fb53f07abcfdd7700a7#r75580623>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF3yl3nX7e2T412jplF8KwmF32Hpg9Ubks5qhyFlgaJpZM4Jo5ec>
> .
>